java-runtime/ftl-runtime/runtime/src/main/java/xyz/block/ftl/runtime/FTLController.java

```diff
@@ -68,14 +70,22 @@ public void onError(Throwable throwable) {
                     waiters = false;
                 }
             }
+            if (!closed) {
+                verbService.getModuleContext(ModuleContextRequest.newBuilder().setModule(moduleName).build(), moduleObserver);
+            }
         }
 
         @Override
         public void onCompleted() {
-            verbService.getModuleContext(ModuleContextRequest.newBuilder().setModule(moduleName).build(), moduleObserver);
+            onError(new RuntimeException("connection closed"));
```

Comment:
nit: an explanation for concluding that an error here must be caused by a connection being closed warrants a comment

Reply:
I guess it is really stream closed rather than connection closed.